emc2/core/model.py

```diff
@@ -663,216 +666,254 @@ def __init__(self, file_path, column_extent,
+            self.ds[self.conv_frac_names[hyd_type]] = xr.DataArray(
+                np.zeros_like(ds["P"].values),
+                dims=('Time', 'bottom_top',
+                      'south_north', 'west_east')).astype('float64')
+            self.ds["q%sc" % hyd_type] = xr.DataArray(
+                self.ds["conv_frac"].values,
+                dims=('Time', 'bottom_top',
+                      'south_north', 'west_east')).astype('float64')
+            self.ds["q%ss" % hyd_type] = ds[self.q_names[hyd_type]]
+            cldfrac2_pl = cldfrac2
+            self.ds[self.strat_frac_names[hyd_type]] = xr.DataArray(
+                cldfrac2_pl,
+                dims=('Time', 'bottom_top',
+                      'south_north', 'west_east')).astype('float64')
+            self.ds[self.N_field[hyd_type]] = ds[
+                self.N_field[hyd_type]].astype('float64') * qn_conversion
+            
+            if parameterization.lower() == "nssl":
+                # Convert from microns to meters
+                self.ds[self.conv_re_fields[hyd_type]] = ds[self.conv_re_fields[hyd_type]].astype('float64') * 1e6
+                self.ds[self.conv_re_fields[hyd_type]].attrs["units"] = "microns"
+                self.ds[self.strat_re_fields[hyd_type]] = ds[self.strat_re_fields[hyd_type]].astype('float64') * 1e6
+                self.ds[self.strat_re_fields[hyd_type]].attrs["units"] = "microns"
+
+        
+        self.ds["QVAPOR"] = ds["QVAPOR"]
+        if parameterization == "nssl":
+            x = ds["QGRAUP"] / ds["QVGRAUPEL"]
+            x = np.where(np.isfinite(x), x, 900.)
+            self.ds["RHO_GRAUPEL"] = xr.DataArray(x, dims=ds["QGRAUP"].dims)
+            x = ds["QHAIL"] / ds["QVHAIL"]
+            x = np.where(np.isfinite(x), x, 900.)
+            self.ds["RHO_HAIL"] = xr.DataArray(x, dims=ds["QGRAUP"].dims)
+            self.ds["RHO_GRAUPEL"] = self.ds["RHO_GRAUPEL"] * (ureg.kg / (ureg.m**3))
+            self.ds["RHO_HAIL"] = self.ds["RHO_HAIL"] * (ureg.kg / (ureg.m**3))
+            self.variable_density = {'gr': "RHO_GRAUPEL",
+                                     'ha': "RHO_HAIL"}
         self.time_dim = "Time"
         self.height_dim = "bottom_top"
         self.model_name = "WRF"
-        self.lat_dim = "XLAT"
-        self.lon_dim = "XLONG"
+        self.lat_dim = "south_north"
+        self.lon_dim = "west_east"
         self.process_conv = False
         wrfin.close()
         for keys in self.ds.keys():
             try:
-                self.ds[keys] = self.ds[keys].drop("XTIME")
+                self.ds[keys] = self.ds[keys].drop_vars("Time")
+            except (KeyError, ValueError):
+                continue
+        for keys in self.ds.keys():
+            try:
+                self.ds[keys] = self.ds[keys].drop_vars("XTIME")
             except (KeyError, ValueError):
                 continue
-        self.ds = xr.Dataset(self.ds)
 
+        self.ds = xr.Dataset(self.ds)
+        if bounding_box is not None:
```

Comment:
It made me able to do the processing without running out of memory. So if we're doing a regional model, we can process on sub-domains then combine the results in the end so that we're aren't exceeding our resource use.